samples/msal-node-samples/ElectronSystemBrowserTestApp/src/CustomLoopbackClient.ts

```diff
@@ -140,4 +139,63 @@ export class CustomLoopbackClient implements ILoopbackClient {
                 });
         });
     }
+
+    /**
+     * Returns URL query string as server auth code response object.
+     */
+    static getDeserializedQueryString(
+        query: string
+    ): ServerAuthorizationCodeResponse {
+        // Check if given query is empty
+        if (!query) {
+            return {};
+        }
+        // Strip the ? symbol if present
+        const parsedQueryString = this.parseQueryString(query);
+        // If ? symbol was not present, above will return empty string, so give original query value
+        const deserializedQueryString: ServerAuthorizationCodeResponse =
+            this.queryStringToObject(
+                parsedQueryString || query
+            );
+        // Check if deserialization didn't work
+        if (!deserializedQueryString) {
+            throw "Unable to deserialize query string";
+        }
+        return deserializedQueryString;
+    }
+
+    /**
+     * Parses query string from given string. Returns empty string if no query symbol is found.
+     * @param queryString
+     */
+    static parseQueryString(queryString: string): string {
```

Comment:
Regex match of https://domain.com/path?.. or https://domain.com?... etc.  might be safer.